### PR TITLE
libatomic_ops: change cmake package name to Atomic_ops

### PR DESCRIPTION
**Pull request by ivmai**
This is to match cmake package/target naming of upstream.

Also, prepare for Conan 2.0.

Specify library name and version:  **libatomic_ops/7.6.14**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libatomic_ops/all/conanfile.py [hunk 1]
```diff
@@ -23,88 +27,78 @@ class Libatomic_opsConan(ConanFile):
...
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "atomic_ops"
-        self.cpp_info.libs = ["atomic_ops_gpl", "atomic_ops"]
+        self.cpp_info.set_property("cmake_file_name", "Atomic_ops")
+        self.cpp_info.set_property("cmake_target_name", "Atomic_ops::Atomic_ops")
```
Comment (SpaceIm) [inline recipes/libatomic_ops/all/conanfile.py:90]: Is there such target upstream?

Comment (ivmai) [reply recipes/libatomic_ops/all/conanfile.py:94]: Yes, https://github.com/ivmai/libatomic_ops/blob/master/CMakeLists.txt#L340
https://github.com/ivmai/libatomic_ops/blob/master/CMakeLists.txt#L143

**Comment by ivmai** [comment]
Unfortunately I don't have access to continuous-integration/jenkins/pr-merge logs to figure out build failure.

File: recipes/libatomic_ops/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libatomic_ops/all/conanfile.py:94]: No, this one is a component in package_info(), and it's name is `Atomic_ops::atomic_ops`

**Comment by SpaceIm** [comment]
> Unfortunately I don't have access to continuous-integration/jenkins/pr-merge logs to figure out build failure.

Do not modify other files than the ones under `recipes/libatomic_ops folder`.

File: recipes/libatomic_ops/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libatomic_ops/all/conanfile.py:94]: see https://github.com/conan-io/conan-center-index/pull/13042#discussion_r975685600

**Comment by ivmai** [comment]
> > Unfortunately I don't have access to continuous-integration/jenkins/pr-merge logs to figure out build failure.
> 
> Do not modify other files than the ones under `recipes/libatomic_ops` folder. Modifications in `bdwgc` recipe must be done in an another PR (same for docs).

Hmm. Docs change can be in another PR, but how I could avoid a change in bdwgc? Do you mean both PRs will be merged together so no warry about inconsistency between?

**Comment by SpaceIm** [comment]
> Do you mean both PRs will be merged together so no warry about inconsistency between?

Indeed, inconsistency can't be avoided here. And even if it was possible, nothing could guarantee consistency in local cache of users, conan is not a distribution but a collection of recipes with different recipe revisions, some of them being incompatible with recipe revision of their dependencies.

File: recipes/libatomic_ops/all/conanfile.py [hunk 2]
```diff
@@ -23,88 +27,82 @@ class Libatomic_opsConan(ConanFile):
...
```
[the remark this answers is not in the document]
Comment (ivmai) [reply recipes/libatomic_ops/all/conanfile.py]: Done

**Comment by ivmai** [comment]
Now it is failed because I provided the patch only for v7.6.14. More work is needed. The patches for v7.6.10 and v7.6.12 should be almost identical except for verison numbers hardcoded in cmake script.

**Comment by ivmai** [comment]
@SpaceIm What's wrong with the build? pkgconfig/atomic_ops.pc.in should exist.
CMake Error: File /home/conan/w/prod/BuildSingleReference/.conan/data/libatomic_ops/7.6.14/_/_/build/fabbd600c30f2c5e046cbca54bb54d4fb9cc5ba0/src/pkgconfig/atomic_ops.pc.in does not exist.
CMake Error at CMakeLists.txt:272 (configure_file):
  configure_file Problem configuring file

**Comment by uilianries** [comment]
@ivmai Please, consider https://github.com/ivmai/conan-center-index/pull/4

**Comment by ivmai** [comment]
@SpaceIm I've fixed all issues, the patch is ready for review.

File: recipes/libatomic_ops/all/conandata.yml [hunk 3]
```diff
@@ -1,10 +1,7 @@
...
-    sha256: "587edf60817f56daf1e1ab38a4b3c729b8e846ff67b4f62a6157183708f099af"
-  "7.6.12":
-    url: "https://github.com/ivmai/libatomic_ops/releases/download/v7.6.12/libatomic_ops-7.6.12.tar.gz"
-    sha256: "f0ab566e25fce08b560e1feab6a3db01db4a38e5bc687804334ef3920c549f3e"
   "7.6.14":
     url: "https://github.com/ivmai/libatomic_ops/releases/download/v7.6.14/libatomic_ops-7.6.14.tar.gz"
     sha256: "390f244d424714735b7050d056567615b3b8f29008a663c262fb548f1802d292"
+patches:
+  "7.6.14":
+    - patch_file: "patches/enable-cmake-7_6_14.patch"
```
Comment (uilianries) [inline recipes/libatomic_ops/all/conandata.yml:7]: please, update the cmake patch fields, https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches-fields

The patch_source is essential in this case, it's a big patch and should come from the upstream.

Comment (ivmai) [reply recipes/libatomic_ops/all/conandata.yml:7]: Done

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**